arviz/data/io_pyro.py

```diff
@@ -57,8 +82,25 @@ def __init__(
                 # model arguments and keyword arguments
                 self._args = self.posterior._args  # pylint: disable=protected-access
                 self._kwargs = self.posterior._kwargs  # pylint: disable=protected-access
+        elif self.num_chains is not None:
+            self.nchains = self.num_chains
+            get_from = None
+            if predictions is not None:
+                get_from = predictions
+            elif posterior_predictive is not None:
+                get_from = posterior_predictive
+            elif prior is not None:
+                get_from = prior
+            if get_from is None:
+                # pylint: disable=line-too-long
+                raise ValueError(
+                    """When constructing InferenceData must have at least
+                                    one of trace, prior, posterior_predictive or predictions."""
+                )
+            aelem = arbitrary_element(get_from)
+            self.ndraws = aelem.shape[0] // self.num_chains
         else:
-            self.nchains = self.ndraws = 0
+            raise ValueError("`num_chains` is needed if trace is not given.")
```

Comment:
I would not raise an error, for example with prior, sampling is generally not performed with multiple chains, from_pyro with only prior or predictions and no num chains should work (even if merging afterwards does not or does not work properly)

Reply:
Then should I set chains and draws to 0?

Reply:
In the interest of not sending warnings when they are not needed, maybe it would be best to set `num_chains=1` in the function definitions. I think it will do the same as the else you are proposing given that num_chains is ignored if posterior is present.